pilot/pkg/networking/plugin/authz/rbac_v2.go

```diff
@@ -0,0 +1,171 @@
+// limitations under the License.
+
+// Package authz converts Istio authorization policies (ServiceRole and AuthorizationPolicy)
+// to corresponding filter config that is used by the envoy RBAC filter to enforce access control to
+// the service co-located with envoy.
+// Currently the config is only generated for sidecar node on inbound HTTP/TCP listener. The generation
+// is controlled by ClusterRbacConfig (a singleton custom resource with cluster scope). User could disable
+// this plugin by either deleting the ClusterRbacConfig or set the ClusterRbacConfig.mode to OFF.
+// Note: ClusterRbacConfig is not created with default istio installation which means this plugin doesn't
+// generate any RBAC config by default.
+//
+// Changes from rbac_v2.go compared to rbac.go:
+// * Deprecate ServiceRoleBinding. Only support two CRDs: ServiceRole and AuthorizationPolicy.
+// * Allow multiple bindings and roles in one CRD, i.e. Authorization.
+// * Support workload selector.
+package authz
+
+import (
+	"strconv"
+	"strings"
+
+	http_config "github.com/envoyproxy/go-control-plane/envoy/config/filter/http/rbac/v2"
+	policyproto "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v2alpha"
+
+	rbacproto "istio.io/api/rbac/v1alpha1"
+	"istio.io/istio/pilot/pkg/model"
+)
+
+// convertRbacRulesToFilterConfigV2 is the successor of convertRbacRulesToFilterConfig, which supports
+// converting AuthorizationPolicy.
+func convertRbacRulesToFilterConfigV2(service *serviceMetadata, option rbacOption) *http_config.RBAC {
+	rbac := &policyproto.RBAC{
+		Action:   policyproto.RBAC_ALLOW,
+		Policies: map[string]*policyproto.Policy{},
+	}
+
+	if option.globalPermissiveMode {
+		// Permissive Mode will be ignored.
+		rbacLog.Debugf("Permissive mode is not supported for service %s.", service.name)
+		// TODO(pitlv2109): Handle permissive mode in the future.
+		return &http_config.RBAC{Rules: rbac}
+	}
+
+	namespace := service.attributes[attrDestNamespace]
+	// Check to make sure we are not dereferencing null pointers.
+	if option.authzPolicies == nil || option.authzPolicies.NamespaceToAuthorizationConfigV2 == nil {
+		return &http_config.RBAC{Rules: rbac}
+	}
+	if _, present := option.authzPolicies.NamespaceToAuthorizationConfigV2[namespace]; !present {
+		return &http_config.RBAC{Rules: rbac}
+	}
+	// Get all AuthorizationPolicy Istio config from this namespace.
+	allAuthzPolicies := option.authzPolicies.NamespaceToAuthorizationConfigV2[namespace].AuthzPolicies
+	// Get all key-value pIstioairs of AuthorizationPolicy name to list of ServiceRoleBindings in this namespace.
+	for _, authzPolicy := range allAuthzPolicies {
+		// If a WorkloadSelector is used in the AuthorizationPolicy config and does not match this service, skip this
+		// AuthorizationPolicy.
+		if authzPolicy.Policy.WorkloadSelector != nil &&
+			!(model.LabelsCollection{service.labels}.IsSupersetOf(authzPolicy.Policy.WorkloadSelector.Labels)) {
```

Comment:
Create a variable for `model.LabelsCollection{service.labels}`, the line is too long.